xds/src/test/java/io/grpc/xds/internal/rbac/engine/CelEvaluationTest.java

```diff
@@ -0,0 +1,318 @@
+        .setIdentExpr(Ident.newBuilder().setName("Condition 1").build())
+        .build();
+    condition2 = Expr.newBuilder()
+        .setIdentExpr(Ident.newBuilder().setName("Condition 2").build())
+        .build();
+    condition3 = Expr.newBuilder()
+        .setIdentExpr(Ident.newBuilder().setName("Condition 3").build())
+        .build();
+    condition4 = Expr.newBuilder()
+        .setIdentExpr(Ident.newBuilder().setName("Condition 4").build())
+        .build();
+    condition5 = Expr.newBuilder()
+        .setIdentExpr(Ident.newBuilder().setName("Condition 5").build())
+        .build();
+    condition6 = Expr.newBuilder()
+        .setIdentExpr(Ident.newBuilder().setName("Condition 6").build())
+        .build();
+    // Set up RBAC policy.
+    policy1 = Policy.newBuilder().setCondition(condition1).build();
+    policy2 = Policy.newBuilder().setCondition(condition2).build();
+    policy3 = Policy.newBuilder().setCondition(condition3).build();
+    policy4 = Policy.newBuilder().setCondition(condition4).build();
+    policy5 = Policy.newBuilder().setCondition(condition5).build();
+    policy6 = Policy.newBuilder().setCondition(condition6).build();
+    // Set up RBAC.
+    rbacAllow = RBAC.newBuilder()
+        .setAction(Action.ALLOW)
+        .putPolicies("Policy 1", policy1)
+        .putPolicies("Policy 2", policy2)
+        .putPolicies("Policy 3", policy3)
+        .build();
+    rbacDeny = RBAC.newBuilder()
+        .setAction(Action.DENY)
+        .putPolicies("Policy 4", policy4)
+        .putPolicies("Policy 5", policy5)
+        .putPolicies("Policy 6", policy6)
+        .build();
+  }
+
+  @Before
+  public void setupEngineSingleRbacAllow() {
+    buildRbac();
+    List<RBAC> rbacList = new ArrayList<>(Arrays.asList(new RBAC[] {rbacAllow}));
+    engine = new AuthorizationEngine<>(ImmutableList.copyOf(rbacList));
+    spyEngine = Mockito.spy(engine);
+    doReturn(ImmutableMap.copyOf(attributes)).when(spyEngine).extractFields(
+        ArgumentMatchers.<EvaluateArgs<ReqT,RespT>>any());
+  }
+
+  @Before
+  public void setupEngineSingleRbacDeny() {
+    buildRbac();
+    List<RBAC> rbacList = new ArrayList<>(Arrays.asList(new RBAC[] {rbacDeny}));
+    engine = new AuthorizationEngine<>(ImmutableList.copyOf(rbacList));
+    spyEngine = Mockito.spy(engine);
+    doReturn(ImmutableMap.copyOf(attributes)).when(spyEngine).extractFields(
+        ArgumentMatchers.<EvaluateArgs<ReqT,RespT>>any());
+  }
+
+  @Before
+  public void setupEngineRbacPair() {
+    buildRbac();
+    List<RBAC> rbacList = new ArrayList<>(Arrays.asList(new RBAC[] {rbacDeny, rbacAllow}));
+    engine = new AuthorizationEngine<>(ImmutableList.copyOf(rbacList));
+    spyEngine = Mockito.spy(engine);
+    doReturn(ImmutableMap.copyOf(attributes)).when(spyEngine).extractFields(
+        ArgumentMatchers.<EvaluateArgs<ReqT,RespT>>any());
+  }
+
+  @Test
+  public void testEvaluateEngineRbacAllow() throws InterpreterException {
```

Comment:
For a test this long, you can add some comments on the purposes of each policy, or try to break them down into small pieces. Otherwise it's a bit hard to keep track of what's tested in each small section.

Reply:
I see. I will split it up to smaller tests then.